### PR TITLE
support custom elasticsearch settings in index level.

### PR DESCRIPTION
**Pull request by JaredTan95**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance
___
### New feature or improvement
- Describe the details and related test reports.

support custom elasticsearch settings in index level.
such as: 
- translog settings: https://www.elastic.co/guide/en/elasticsearch/reference/7.4/index-modules-translog.html

### CODE REVIEW

File: oap-server/server-bootstrap/src/main/resources/application.yml [hunk 1]
```diff
@@ -119,6 +122,9 @@ storage:
     # instead, give more query criteria (e.g. service id or time range), to narrow the query results.
     # see https://www.elastic.co/guide/en/elasticsearch/guide/current/pagination.html for more information
     indexMaxResultWindow: ${SW_STORAGE_ES_INDEX_MAX_RESULT_WINDOW:5000}
+#    advanced:
+#      index.translog.durability: "async"
+#      index.translog.sync_interval: "120s"
```
Comment (wu-sheng) [inline oap-server/server-bootstrap/src/main/resources/application.yml]: According to the official document, `index.translog.sync_interval` default value is 5s,  `index.translog.durability` is `request`. But you set the different, are these two recommended by you? If so, it should be active configs in default. 

And you miss to update the documentation of storage.

Comment (wu-sheng) [reply oap-server/server-bootstrap/src/main/resources/application.yml]: The default values are referring from https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-translog.html#_translog_settings

File: docker/oap/docker-entrypoint.sh [hunk 2]
```diff
@@ -119,6 +119,7 @@ cat <<EOT >> ${var_application_file}
     resultWindowMaxSize: \${SW_STORAGE_ES_QUERY_MAX_WINDOW_SIZE:10000}
     metadataQueryMaxSize: \${SW_STORAGE_ES_QUERY_MAX_SIZE:5000}
     segmentQueryMaxSize: \${SW_STORAGE_ES_QUERY_SEGMENT_SIZE:200}
+    topNReportPeriod: ${SW_CORE_TOPN_REPORT_PERIOD:10}
```
Comment (kezhenxu94) [inline docker/oap/docker-entrypoint.sh]:
```suggestion
    topNReportPeriod: \${SW_CORE_TOPN_REPORT_PERIOD:10}
```

`SW_CORE_`?

File: oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java [hunk 3]
```diff
@@ -43,14 +42,17 @@
     protected final int indexShardsNumber;
     protected final int indexReplicasNumber;
     protected final int indexRefreshInterval;
+    protected final Map<String,Object> advanced;
     protected final ColumnTypeEsMapping columnTypeEsMapping;
 
-    public StorageEsInstaller(ModuleManager moduleManager, int indexShardsNumber, int indexReplicasNumber, int indexRefreshInterval) {
+    public StorageEsInstaller(ModuleManager moduleManager, int indexShardsNumber, int indexReplicasNumber, int indexRefreshInterval,
```
Comment (kezhenxu94) [inline oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java]: The constructor arguments are becoming too long, consider passing the `config` directly and get the config items inside the constructor, like what the ES7 version does

File: oap-server/server-bootstrap/src/main/resources/application.yml (hunk 1 above)
Comment (kezhenxu94) [reply oap-server/server-bootstrap/src/main/resources/application.yml]: > According to the official document, `index.translog.sync_interval` default value is 5s, `index.translog.durability` is `request`. But you set the different, are these two recommended by you? If so, it should be active configs in default.
> 
> And you miss to update the documentation of storage.

same question here

File: docs/en/setup/backend/backend-storage.md [hunk 4]
```diff
@@ -103,12 +104,24 @@ storage:
...
+    advanced: ${SW_STORAGE_ES_ADVANCED:""}
 ```
 
 
 ### Data TTL
 TTL in ElasticSearch overrides the settings of core, read [ElasticSearch section in TTL document](ttl.md#elasticsearch-6-storage-ttl)
 
+### Advanced Configurations For Elasticsearch Index
+You can add advanced configurations in `JSON` format to set `elasticsearch index settings`.
+For example, set [translog](https://www.elastic.co/guide/en/elasticsearch/reference/master/index-modules-translog.html) settings:
```
Comment (wu-sheng) [inline docs/en/setup/backend/backend-storage.md]: I think this doc ref will misguide people. From my understanding, all parameters in [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html) should be supported, right?

Comment (JaredTan95) [reply docs/en/setup/backend/backend-storage.md]: Yes.

Comment (JaredTan95) [reply docs/en/setup/backend/backend-storage.md]: Need to add this ref https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html ?

Comment (wu-sheng) [reply docs/en/setup/backend/backend-storage.md]: Then you should ref to the config page in the document, your example could be same. Such as
___
You can add advanced configurations in `JSON` format to set `ElasticSearch index settings` by following [ElasticSearch doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html)

Comment (JaredTan95) [reply docs/en/setup/backend/backend-storage.md]: Thanks, I will add it.

**Pull request merged**